src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2785,17 +2790,34 @@ internal bool ShouldSuppressAnalysisWarningsForRequiresUnreferencedCode ()
 
 			IMemberDefinition suppressionContextMember = currentOrigin.SuppressionContextMember;
 			if (suppressionContextMember != null &&
-				Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (suppressionContextMember))
+				(Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (suppressionContextMember) ||
+				(suppressionContextMember.DeclaringType != null &&
+				Annotations.HasRequiresUnreferencedCodeOnTypeHierarchy (suppressionContextMember.DeclaringType))))
 				return true;
 
 			IMemberDefinition originMember = currentOrigin.MemberDefinition;
 			if (suppressionContextMember != originMember && originMember != null &&
-				Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (originMember))
+				(Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (originMember) ||
+				(originMember.DeclaringType != null &&
+				Annotations.HasRequiresUnreferencedCodeOnTypeHierarchy (originMember.DeclaringType))))
 				return true;
 
 			return false;
 		}
 
+		internal void CheckAndReportRequiresUnreferencedCode (TypeDefinition type)
+		{
+			var currentOrigin = _scopeStack.CurrentScope.Origin;
+
+			// If the caller of a type is already marked with `RequiresUnreferencedCodeAttribute` a new warning should not
+			// be produced for the callee.
+			if (ShouldSuppressAnalysisWarningsForRequiresUnreferencedCode ())
+				return;
+
+			if (Annotations.TryGetLinkerAttribute (_context.TryResolve (type.BaseType), out RequiresUnreferencedCodeAttribute requiresUnreferencedCodeOnTypeHierarchy))
+				ReportRequiresUnreferencedCode (type.BaseType.GetDisplayName (), requiresUnreferencedCodeOnTypeHierarchy, currentOrigin);
```

Comment:
I think the origin here is potentially wrong. The warning about the base type should be always reported on the derived type, but the scope stack will probably point to the location of a reference to the derived type.

Reply:
I removed the Suppress function call all along, for this specific scenario I think the normal suppression rules don't apply.
Added a test case where I suppress IL2109 by checking that the parent class has RUC, which I think is the actual suppression that applies for IL2109